### PR TITLE
mate.libmatekbd: build with meson, fix cross compilation

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/mate/libmatekbd/default.nix [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
-    gtk3-x11
+    glib
+    gtk3
     libxklavier
   ];
 
-  enableParallelBuilding = true;
-
-  passthru.updateScript = mateUpdateScript { inherit pname; };
+  passthru.updateScript = nix-update-script { };
```
Comment (SuperSandro2000) [inline pkgs/desktops/mate/libmatekbd/default.nix:44]: Don't they do odd even stable?

Comment (NickCao) [reply pkgs/desktops/mate/libmatekbd/default.nix:44]: Thanks for reminder. Seems they do use odd-unstable, do nix-update-script support this?

File: pkgs/desktops/mate/libmatekbd/default.nix [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
+    meson
+    ninja
     pkg-config
-    gettext
+    glib
+    gobject-introspection
   ];
 
   buildInputs = [
-    gtk3-x11
```
Comment (bobby285271) [inline pkgs/desktops/mate/libmatekbd/default.nix:26]: I think `gdk/gdkx.h` requires gtk3-x11 on non-linux, we can see if ofborg builds darwin, ~~maybe we should just mark mate packages linux only~~ :shrug:

File: pkgs/desktops/mate/libmatekbd/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/mate/libmatekbd/default.nix:44]: `gitUpdater` supports it.

Comment (romildo) [reply pkgs/desktops/mate/libmatekbd/default.nix:44]: `gitUpdater { odd-unstable = true; };`

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 330713` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>51 packages built:</summary>
  <ul>
    <li>ayatana-indicator-display</li>
    <li>ayatana-indicator-session</li>
    <li>blueberry</li>
    <li>cinnamon-common</li>
    <li>cinnamon-gsettings-overrides</li>
    <li>cinnamon-screensaver</li>
    <li>cinnamon-session</li>
    <li>hypnotix</li>
    <li>lightdm-slick-greeter</li>
    <li>mate.libmatekbd</li>
    <li>mate.marco</li>
    <li>mate.mate-applets</li>
    <li>mate.mate-control-center</li>
    <li>mate.mate-gsettings-overrides</li>
    <li>mate.mate-indicator-applet</li>
    <li>mate.mate-media</li>
    <li>mate.mate-netbook</li>
    <li>mate.mate-notification-daemon</li>
    <li>mate.mate-panel</li>
    <li>mate.mate-panel-with-applets</li>
    <li>mate.mate-power-manager</li>
    <li>mate.mate-screensaver</li>
    <li>mate.mate-sensors-applet</li>
    <li>mate.mate-session-manager</li>
    <li>mate.mate-settings-daemon</li>
    <li>mate.mate-settings-daemon-wrapped</li>
    <li>mate.mate-tweak</li>
    <li>mate.mate-tweak.dist</li>
    <li>mate.mate-utils</li>
    <li>mate.mate-wayland-session</li>
    <li>nemo</li>
    <li>nemo-fileroller</li>
    <li>nemo-python</li>
    <li>nemo-with-extensions</li>
    <li>nemo.dev</li>
    <li>pix</li>
    <li>plata-theme</li>
    <li>python311Packages.python-xapp</li>
    <li>python312Packages.python-xapp</li>
    <li>sticky</li>
    <li>timeshift</li>
    <li>timeshift-minimal</li>
    <li>timeshift-unwrapped</li>
    <li>warpinator</li>
    <li>xapp</li>
    <li>xapp.dev</li>
    <li>xed-editor</li>
    <li>xmonad_log_applet_mate</li>
    <li>xplayer</li>
    <li>xreader</li>
    <li>xviewer</li>
  </ul>
</details>

**Review by romildo — APPROVED** [review]
LGTM

**Comment by bobby285271** [comment]
We probably should report to upstream about missing glib-compile-schemas post install script #331317 :upside_down_face: 
<sup><sub>(Though I don't have motivation to switch more mate packages to meson due to the bad experience in #289062)</sub></sup>

**Pull request merged**